lib/core/utils/matches.js

```diff
@@ -128,6 +128,7 @@ function convertAttributes(atts) {
     return {
       key: attributeKey,
       value: attributeValue,
+      type: typeof att.value === 'undefined' ? 'attrExist' : 'attrValue',
```

Comment:
This was added to tell the difference between the selectors `[href]` and `[href=""]` (both returned the same structure of `value: ''` even though one is only checking existence of the attribute and the other is checking for an empty string).